src/t8_schemes/t8_new_feature/t8_subelements/t8_subelements_quad_cxx.cxx

```diff
@@ -0,0 +1,1325 @@
+   *        | elem      |              | /   6   \ |           consider sub_id 3:                                                x -> second subelement on the upper face
+   *        + - - - - - x              x - - - - - x
+   *              f1                         0
+   *           
+   * We will use the binary representation to determine the location of the given subelement. 
+   * 
+   * We need to know: 
+   *     i)   the face number of the first vertex. Values: {0,1,2,3}
+   *     ii)  whether this face is split in half. Values {0,1}
+   *     iii) if the subelement is the first or second subelement at the face. Values: {0,1}
+   * 
+   * These informations are then saved in the location array which will be used by the element_vertex function, 
+   * to automatically determine the vertex coordinates of the given subelement. 
+   * 
+   * The location array for the above example would be {1,1,1} (upper face, split = true, second subelement at the upper face). */
+  
+  /* 1) convert the subelement type from a decimal to a binary representation */
+  int type = pquad_w_sub->subelement_type;
+  int binary_array[P4EST_FACES] = {};
+
+  int i;
+
+  for (i = 0; i < P4EST_FACES; i++) { /* need an array with 4 elements to store all subelement types of the quad scheme from 1 to 15 ({0,0,0,1} to {1,1,1,1}) */
+    if (type >= pow (2, (P4EST_FACES - 1) - i)) {
+      binary_array[i] = 1;
+      type -= pow (2, (P4EST_FACES - 1) - i);
+    }
+  } /* we now got a binary represenation of the subelement type, bitwise stored in an array */
+
+  /* 2) rearrange the binary representation to be in clockwise order */
+  int binary_array_temp[P4EST_FACES] = {};
+
+  int j;
+
+  for (j = 0; j < P4EST_FACES; j++) { /* copying the binary array */
+    binary_array_temp[j] = binary_array[j];
+  }
+
+  binary_array[0] = binary_array_temp[0]; /* f3 <- f3 */
+  binary_array[1] = binary_array_temp[3]; /* f2 <- f0 */
+  binary_array[2] = binary_array_temp[1]; /* f1 <- f2 */
+  binary_array[3] = binary_array_temp[2]; /* f0 <- f1 */
+
+  /* 3) use the rearranged binary representation, and the sub_id to determine the location of the subelement and store these information in an array */
+  /* 3.1) location[0] -> the face_number, the subelement is adjacent to */
+  /* 3.2) location[1] -> dummy, if the face is split or not */
+  /* 3.3) location[2] -> is the subelement the first or second subelement of the face (always the first, if the face is not split) */
+  T8_ASSERT (pquad_w_sub->subelement_id < pquad_w_sub->num_subelement_ids);
+
+  int sub_id = pquad_w_sub->subelement_id;
+  int sub_face_id = 1; /* initialize the sub_face_id as the second subelement of a face and change that if it is not true */
+  int face_number = 0;
+  int split;
+
+  int k = 0;
+
+  if (sub_id == 0) {
+    sub_face_id = 0; /* in this case, the subelement is the first element of the face */
+  }
+
+  while (sub_id >= 0 && k < P4EST_FACES) {
```

Comment:
You can (and should) use a for loop here
```suggestion
  for (k = 0; sub_id >= 0 && k < P4EST_FACES;++k) {
```

Reply:
done

Reply:
Additional adjustements had to be made here, so I changed the whole structure of this part to be cleaner and better to understand.